app/gui/src/ide/initializer.rs

```diff
@@ -109,19 +106,17 @@ impl Initializer {
         match &self.config.backend {
             ProjectManager { endpoint } => {
                 let project_manager = self.setup_project_manager(endpoint).await?;
-                let project_name = self.config.project_name.clone();
-                let controller = controller::ide::Desktop::new_with_opened_project(
-                    project_manager,
-                    project_name,
-                )
-                .await?;
+                let maybe_project_name = self.config.project_name.clone();
+                let controller =
+                    controller::ide::Desktop::new(project_manager, maybe_project_name).await?;
                 Ok(Rc::new(controller))
             }
             LanguageServer { json_endpoint, binary_endpoint, namespace } => {
                 let json_endpoint = json_endpoint.clone();
                 let binary_endpoint = binary_endpoint.clone();
                 let namespace = namespace.clone();
-                let project_name = self.config.project_name.clone();
+                // unwrap is safe here, as BackendService constructor made this check
```

Comment:
Please write a full sentence in the comment. Also, it while this should never happen, it would still be better to use an `expect` with a meaningful error message in case a bug somewhere triggers it at some point.

Reply:
Noted, comment removed as part of further refactoring: `project_name` now is a required part of `LanguageServer`, so no unwrap is needed.